Sources/Prometheus/PrometheusCollectorRegistry.swift

```diff
@@ -790,59 +991,163 @@ public final class PrometheusCollectorRegistry: Sendable {
 
     // MARK: Emitting
 
+    private let bufferBox = NIOLockedValueBox([UInt8]())
+
+    /// Resets the internal buffer used by ``emitToString()`` and ``emitToBuffer()``.
+    ///
+    /// Forces the buffer capacity back to 0, which will trigger re-calibration on the next emission call.
+    /// This is useful when the registry's metric composition has changed significantly and you want to
+    /// optimize buffer size for the new workload.
+    ///
+    /// - Note: Thread-safe. Does not affect ``emit(into:)`` calls which use external buffers
+    public func resetInternalBuffer() {
+        bufferBox.withLockedValue { buffer in
+            // Resets capacity to 0, forcing re-calibration
+            buffer.removeAll()
+        }
+    }
+
+    /// Returns the current capacity of the internal buffer used by ``emitToString()`` and ``emitToBuffer()``.
+    ///
+    /// The capacity represents the allocated memory size, not the current content length. A capacity of 0
+    /// indicates the buffer will auto-size on the next emission call. The capacity may grow over time as
+    /// the registry's output requirements increase.
+    ///
+    /// - Returns: The current buffer capacity in bytes
+    /// - Note: Thread-safe. Primarily useful for testing and monitoring buffer behavior
+    public func internalBufferCapacity() -> Int {
+        return bufferBox.withLockedValue { buffer in
+            buffer.capacity
+        }
+    }
+
+    /// Emits all registered metrics in Prometheus text format as a String.
+    ///
+    /// Uses an internal buffer that auto-sizes on first call to find optimal initial capacity. The buffer
+    /// may resize during the registry's lifetime if output grows significantly. Subsequent calls reuse the
+    /// established capacity, clearing content but preserving the initially allocated memory.
+    ///
+    /// - Returns: A String containing all registered metrics in Prometheus text format
+    /// - Note: Thread-safe. Use ``resetInternalBuffer()`` to force recalibration
+    /// - SeeAlso: ``emitToBuffer()`` for raw UTF-8 bytes, ``emit(into:)`` for custom buffer
+    public func emitToString() -> String {
+        return bufferBox.withLockedValue { buffer in
+            guard buffer.capacity == 0 else {
```

Comment:
I'd rather have a plain if else here because this isn't really guard + early exit but _actual logic_ in the else case; which is better with an if/else

Reply:
Turns out the formatting `curl -s --retry 3 https://raw.githubusercontent.com/swiftlang/github-workflows/refs/heads/main/.github/workflows/scripts/check-swift-format.sh | bash`...  reverts it again 🤷‍♀️

Reply:
It changes if to guard, wat! Our rules are overdoing it clearly then… but let’s not hold up the PR by this